internal/cas/fetch.go

```diff
@@ -69,3 +69,28 @@ func FetchManifest(ctx context.Context, target oras.Target, reference string, p
 	defer rc.Close()
 	return content.ReadAll(rc, desc)
 }
+
+// FetchBlob fetches the blob content of reference from blob store.
+func FetchBlob(ctx context.Context, blob oras.Target, reference string) ([]byte, error) {
+	rf := blob.(registry.ReferenceFetcher)
+	desc, rc, err := rf.FetchReference(ctx, reference)
+	if err != nil {
+		return nil, err
+	}
+	defer rc.Close()
+	return content.ReadAll(rc, desc)
+}
+
+// BlobTarget returns a ORAS Target with a no-op Tag method wrapping the
+// provided blob store b.
+func BlobTarget(b registry.BlobStore) oras.Target {
+	return blobTarget{b}
+}
+
+type blobTarget struct {
+	registry.BlobStore
+}
+
+func (blobTarget) Tag(ctx context.Context, desc ocispec.Descriptor, reference string) error {
+	return nil
```

Comment:
This is probably obvious to most, but to me it's confusing... maybe a comment explaining why it's a do-nothing function?

Reply:
Hi @nima, thanks for your review. 

Basically this `Tag` function is an implementation of `Target.Tag()` in order to make blobTarget as a ORAS Target. In func `FetchBlob`, we will call `FetchReference` to get the blob content. However, if we use Repository as the `Target`, it will go to the manifest workflow (ref https://github.com/oras-project/oras-go/blob/main/registry/remote/repository.go#L257) which will cause the wrong result. So we have to use a `BlobStore` here as the Target. But  `BlobStore` does not have a `Tag` method (ref https://github.com/oras-project/oras-go/blob/main/registry/repository.go#L68). Therefore, this no-op method is added in order to make `blobTarget` as a `Target`.

I just added some comments in the code. Thanks for pointing this out.